Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -680,6 +611,98 @@ public class PBXProjGenerator {
         }
 
         try target.postbuildScripts.forEach(generateBuildScript)
+        
+        let configs: [ObjectReference<XCBuildConfiguration>] = project.configs.map { config in
+            var buildSettings = project.getTargetBuildSettings(target: target, config: config)
+            
+            // automatically set INFOPLIST_FILE path
+            if !project.targetHasBuildSetting("INFOPLIST_FILE", basePath: project.basePath, target: target, config: config) {
+                if searchForPlist {
+                    plistPath = getInfoPlist(target.sources)
+                    searchForPlist = false
+                }
+                if let plistPath = plistPath {
+                    buildSettings["INFOPLIST_FILE"] = plistPath.byRemovingBase(path: project.basePath)
+                }
+            }
+            
+            // automatically calculate bundle id
+            if let bundleIdPrefix = project.options.bundleIdPrefix,
+                !project.targetHasBuildSetting("PRODUCT_BUNDLE_IDENTIFIER", basePath: project.basePath, target: target, config: config) {
+                let characterSet = CharacterSet.alphanumerics.union(CharacterSet(charactersIn: "-.")).inverted
+                let escapedTargetName = target.name
+                    .replacingOccurrences(of: "_", with: "-")
+                    .components(separatedBy: characterSet)
+                    .joined(separator: "")
+                buildSettings["PRODUCT_BUNDLE_IDENTIFIER"] = bundleIdPrefix + "." + escapedTargetName
+            }
+            
+            // automatically set test target name
+            if target.type == .uiTestBundle,
+                !project.targetHasBuildSetting("TEST_TARGET_NAME", basePath: project.basePath, target: target, config: config) {
+                for dependency in target.dependencies {
+                    if dependency.type == .target,
+                        let dependencyTarget = project.getTarget(dependency.reference),
+                        dependencyTarget.type == .application {
+                        buildSettings["TEST_TARGET_NAME"] = dependencyTarget.name
+                        break
+                    }
+                }
+            }
+            
+            // objc linkage
+            let anyDependencyRequiresObjCLinking = targetDependencies.contains { dependency in
```

Comment:
This could be quite performance intensive

Reply:
All searching of sources was removed, and it now just sets a flag once as soon as `anyDependencyRequiresObjCLinking` would be true. Performance shouldn't be an issue anymore.